In [3]:
import pandas as pd
import os

In [4]:
os.getcwd()

'C:\\Users\\Layashree S\\APPLICATION OF SOFTWARE'

In [5]:
os.chdir("C:\QT In Management")

In [6]:
os.getcwd()

'C:\\QT In Management'

In [7]:
pip install openpyxl

Note: you may need to restart the kernel to use updated packages.


In [8]:
# import the ecommerce data file
df_order = pd.read_excel('Ecommerce_data.xlsx', sheet_name='Orders')
df_return = pd.read_excel('Ecommerce_data.xlsx', sheet_name='Returns')
df_people = pd.read_excel('Ecommerce_data.xlsx', sheet_name='People')

### QUESTION - 1 What is the total return value of the orders that are returned??

In [9]:
df_return

Returned                  Order ID         Region_r
0         Yes  CA-2012-SA20830140-41210       Central US
1         Yes  IN-2012-PB19210127-41259     Eastern Asia
2         Yes  CA-2012-SC20095140-41174       Central US
3         Yes    IN-2015-JH158207-42140          Oceania
4         Yes    IN-2014-LC168857-41747          Oceania
...       ...                       ...              ...
1074      Yes   IN-2014-DA1345058-41769    Southern Asia
1075      Yes  US-2013-HG14845140-41530       Eastern US
1076      Yes   US-2013-SJ2021582-41543  Central America
1077      Yes  CA-2015-EB13870140-42269       Eastern US
1078      Yes    EG-2014-PM894038-41846     North Africa

[1079 rows x 3 columns]

In [10]:
# Merging order sheet and return sheet into single data frame
merged_df = pd.merge(df_order, df_return, on='Order ID', how='inner')

In [11]:
list(merged_df)

['Row ID',
 'Order ID',
 'Order Date',
 'Ship Date',
 'Ship Mode',
 'Customer ID',
 'Customer Name',
 'Segment',
 'Postal Code',
 'City',
 'State',
 'Country',
 'Region',
 'Market',
 'Product ID',
 'Category',
 'Sub-Category',
 'Product Name',
 'Sales',
 'Quantity',
 'Discount',
 'Profit',
 'Shipping Cost',
 'Order Priority',
 'Unnamed: 24',
 'Unnamed: 25',
 'Returned',
 'Region_r']

In [12]:
# creating the data frame based on orders which are retured
returned_orders = merged_df[merged_df['Returned'] == 'Yes']

In [13]:
#calculating the total retun value
total_reurn_sales_value = returned_orders['Sales'].sum()
total_reurn_sales_value

525932.22736

#### Total value of the orders that are returned is 525932.22736

### QUESTION - 2 Total return value of the orders by region wise

In [14]:
# calculating the total return value by region wise uing groupby 
returned_orders.groupby('Region_r')['Sales'].sum()

Region_r
Caribbean             9679.32828
Central Africa        4038.48000
Central America      44334.79916
Central Asia           359.03100
Central US           23306.14500
Eastern Africa        2026.44000
Eastern Asia         47798.04300
Eastern Canada        2639.91000
Eastern Europe        9501.21000
Eastern US           23477.32600
North Africa         10193.37000
Northern Europe      17768.64000
Oceania              46943.33100
South America        29845.11212
Southeastern Asia    36967.09830
Southern Africa       7321.86000
Southern Asia        25159.36500
Southern Europe      34226.34300
Southern US          17024.59700
Western Africa        3928.15800
Western Asia         11737.63800
Western Canada         676.50000
Western Europe       72669.48450
Western US           44310.01800
Name: Sales, dtype: float64

### QUESTION - 3 What is the total returned value by category and sub category wise using product ID

In [15]:
list(returned_orders)

['Row ID',
 'Order ID',
 'Order Date',
 'Ship Date',
 'Ship Mode',
 'Customer ID',
 'Customer Name',
 'Segment',
 'Postal Code',
 'City',
 'State',
 'Country',
 'Region',
 'Market',
 'Product ID',
 'Category',
 'Sub-Category',
 'Product Name',
 'Sales',
 'Quantity',
 'Discount',
 'Profit',
 'Shipping Cost',
 'Order Priority',
 'Unnamed: 24',
 'Unnamed: 25',
 'Returned',
 'Region_r']

In [16]:
# creating the category variable from product ID 
returned_orders['Category_1'] = returned_orders['Product ID'].str[:3]

In [17]:
# Calculating the total return value by category wise
returned_orders.groupby('Category_1')['Sales'].sum()

Category_1
FUR    168015.06520
OFF    157195.10930
TEC    200722.05286
Name: Sales, dtype: float64

In [18]:
#creating the sub-category variable from product ID 
returned_orders['sub_Category_1'] = returned_orders['Product ID'].str[4:7]

In [19]:
## Calculating the total return value by sub-category wise
returned_orders.groupby('sub_Category_1')['Sales'].sum()

sub_Category_1
AC-    33991.09180
AP-    32975.10160
AR-    16221.50620
BI-    23571.83300
BO-    60710.99170
CH-    57439.11790
CO-    65145.83696
EN-     6784.22590
FA-     3098.36380
FU-    16581.79860
LA-     3349.15170
MA-    32550.62610
PA-     9928.68380
PH-    69034.49800
ST-    52938.56010
SU-     8327.68320
TA-    33283.15700
Name: Sales, dtype: float64

### QUESTION - 4 What is the total number of products returned whose unit value is more than 100 dolar

In [38]:
#creating unitvalue variable into data frame
returned_orders['unit_price'] = returned_orders[ 'Sales']/returned_orders['Quantity']
returned_orders

Row ID                  Order ID Order Date  Ship Date       Ship Mode  \
0      30191  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
1      30190  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
2      30187  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
3      30193  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
4      30189  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
...      ...                       ...        ...        ...             ...   
2215   35040  CA-2014-WB21850140-41654 2014-01-15 2014-01-21  Standard Class   
2216   35038  CA-2014-WB21850140-41654 2014-01-15 2014-01-21  Standard Class   
2217      25   US-2013-SJ2021582-41543 2013-09-26 2013-09-29     First Class   
2218   36974  CA-2012-ZC21910140-41271 2012-12-28 2013-01-04  Standard Class   
2219   41315    EG-2014-PM894038-41846 2014-07-26 2014-07-30  Standard Class   

       Customer ID     Customer Name    Segment  Postal Code           City  \
0      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
1      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
2      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
3      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
4      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
...            ...               ...        ...          ...            ...   
2215  WB-218501406     William Brown   Consumer      10011.0  New York City   
2216  WB-218501406     William Brown   Consumer      10011.0  New York City   
2217    SJ-2021582      Sarah Jordon   Consumer          NaN     Hermosillo   
2218  ZC-219101402  Zuschuss Carroll   Consumer      60610.0        Chicago   
2219     PM-894038    Paul MacIntyre   Consumer          NaN     Alexandria   

      ... Shipping Cost Order Priority Unnamed: 24 Unnamed: 25 Returned  \
0     ...       725.570       Critical         NaN         NaN      Yes   
1     ...       627.270       Critical         NaN         NaN      Yes   
2     ...       458.970       Critical         NaN         NaN      Yes   
3     ...       203.920       Critical         NaN         NaN      Yes   
4     ...        25.480       Critical         NaN         NaN      Yes   
...   ...           ...            ...         ...         ...      ...   
2215  ...         2.670         Medium         NaN         NaN      Yes   
2216  ...         1.690         Medium         NaN         NaN      Yes   
2217  ...         1.085         Medium         NaN         NaN      Yes   
2218  ...         5.290            Low         NaN         NaN      Yes   
2219  ...         1.050         Medium         NaN         NaN      Yes   

             Region_r Category_1 sub_Category_1  unit_value  unit_price  
0        Eastern Asia        FUR            TA-     857.580     857.580  
1        Eastern Asia        FUR            BO-     439.500     439.500  
2        Eastern Asia        FUR            TA-     452.010     452.010  
3        Eastern Asia        FUR            CH-     176.430     176.430  
4        Eastern Asia        TEC            MA-      49.440      49.440  
...               ...        ...            ...         ...         ...  
2215       Eastern US        OFF            BI-      16.512      16.512  
2216       Eastern US        OFF            EN-       4.130       4.130  
2217  Central America        FUR            FU-      20.124      20.124  
2218       Central US        FUR            FU-      12.992      12.992  
2219     North Africa        OFF            AR-      25.830      25.830  

[2220 rows x 32 columns]

In [39]:
returned_orders

Row ID                  Order ID Order Date  Ship Date       Ship Mode  \
0      30191  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
1      30190  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
2      30187  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
3      30193  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
4      30189  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
...      ...                       ...        ...        ...             ...   
2215   35040  CA-2014-WB21850140-41654 2014-01-15 2014-01-21  Standard Class   
2216   35038  CA-2014-WB21850140-41654 2014-01-15 2014-01-21  Standard Class   
2217      25   US-2013-SJ2021582-41543 2013-09-26 2013-09-29     First Class   
2218   36974  CA-2012-ZC21910140-41271 2012-12-28 2013-01-04  Standard Class   
2219   41315    EG-2014-PM894038-41846 2014-07-26 2014-07-30  Standard Class   

       Customer ID     Customer Name    Segment  Postal Code           City  \
0      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
1      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
2      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
3      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
4      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
...            ...               ...        ...          ...            ...   
2215  WB-218501406     William Brown   Consumer      10011.0  New York City   
2216  WB-218501406     William Brown   Consumer      10011.0  New York City   
2217    SJ-2021582      Sarah Jordon   Consumer          NaN     Hermosillo   
2218  ZC-219101402  Zuschuss Carroll   Consumer      60610.0        Chicago   
2219     PM-894038    Paul MacIntyre   Consumer          NaN     Alexandria   

      ... Shipping Cost Order Priority Unnamed: 24 Unnamed: 25 Returned  \
0     ...       725.570       Critical         NaN         NaN      Yes   
1     ...       627.270       Critical         NaN         NaN      Yes   
2     ...       458.970       Critical         NaN         NaN      Yes   
3     ...       203.920       Critical         NaN         NaN      Yes   
4     ...        25.480       Critical         NaN         NaN      Yes   
...   ...           ...            ...         ...         ...      ...   
2215  ...         2.670         Medium         NaN         NaN      Yes   
2216  ...         1.690         Medium         NaN         NaN      Yes   
2217  ...         1.085         Medium         NaN         NaN      Yes   
2218  ...         5.290            Low         NaN         NaN      Yes   
2219  ...         1.050         Medium         NaN         NaN      Yes   

             Region_r Category_1 sub_Category_1  unit_value  unit_price  
0        Eastern Asia        FUR            TA-     857.580     857.580  
1        Eastern Asia        FUR            BO-     439.500     439.500  
2        Eastern Asia        FUR            TA-     452.010     452.010  
3        Eastern Asia        FUR            CH-     176.430     176.430  
4        Eastern Asia        TEC            MA-      49.440      49.440  
...               ...        ...            ...         ...         ...  
2215       Eastern US        OFF            BI-      16.512      16.512  
2216       Eastern US        OFF            EN-       4.130       4.130  
2217  Central America        FUR            FU-      20.124      20.124  
2218       Central US        FUR            FU-      12.992      12.992  
2219     North Africa        OFF            AR-      25.830      25.830  

[2220 rows x 32 columns]

In [42]:
#calculating the count of return products habing unit value more than 100 dolar
returned_orders[returned_orders['unit_price']>100].count().head(1)

Row ID    473
dtype: int64

#### The total number of products returned whose unit value is > 100 dollar is 473

### QUESTION -5 How many orders delivered on same day, 7th day, 15th day and more than 30 days

In [23]:
merged_df['days_taken']= merged_df['Ship Date']-merged_df['Order Date']
merged_df

Row ID                  Order ID Order Date  Ship Date       Ship Mode  \
0      30191  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
1      30190  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
2      30187  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
3      30193  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
4      30189  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
...      ...                       ...        ...        ...             ...   
2215   35040  CA-2014-WB21850140-41654 2014-01-15 2014-01-21  Standard Class   
2216   35038  CA-2014-WB21850140-41654 2014-01-15 2014-01-21  Standard Class   
2217      25   US-2013-SJ2021582-41543 2013-09-26 2013-09-29     First Class   
2218   36974  CA-2012-ZC21910140-41271 2012-12-28 2013-01-04  Standard Class   
2219   41315    EG-2014-PM894038-41846 2014-07-26 2014-07-30  Standard Class   

       Customer ID     Customer Name    Segment  Postal Code           City  \
0      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
1      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
2      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
3      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
4      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
...            ...               ...        ...          ...            ...   
2215  WB-218501406     William Brown   Consumer      10011.0  New York City   
2216  WB-218501406     William Brown   Consumer      10011.0  New York City   
2217    SJ-2021582      Sarah Jordon   Consumer          NaN     Hermosillo   
2218  ZC-219101402  Zuschuss Carroll   Consumer      60610.0        Chicago   
2219     PM-894038    Paul MacIntyre   Consumer          NaN     Alexandria   

      ... Quantity Discount    Profit Shipping Cost Order Priority  \
0     ...        2      0.0  720.3600       725.570       Critical   
1     ...        5      0.0  153.7500       627.270       Critical   
2     ...        3      0.0  311.8500       458.970       Critical   
3     ...        5      0.0  114.6000       203.920       Critical   
4     ...        3      0.0   68.2200        25.480       Critical   
...   ...      ...      ...       ...           ...            ...   
2215  ...        3      0.2   17.3376         2.670         Medium   
2216  ...        4      0.0    7.5992         1.690         Medium   
2217  ...        9      0.4  -75.5640         1.085         Medium   
2218  ...        3      0.6  -50.6688         5.290            Low   
2219  ...        1      0.0    9.0300         1.050         Medium   

     Unnamed: 24 Unnamed: 25 Returned         Region_r  days_taken  
0            NaN         NaN      Yes     Eastern Asia      3 days  
1            NaN         NaN      Yes     Eastern Asia      3 days  
2            NaN         NaN      Yes     Eastern Asia      3 days  
3            NaN         NaN      Yes     Eastern Asia      3 days  
4            NaN         NaN      Yes     Eastern Asia      3 days  
...          ...         ...      ...              ...         ...  
2215         NaN         NaN      Yes       Eastern US      6 days  
2216         NaN         NaN      Yes       Eastern US      6 days  
2217         NaN         NaN      Yes  Central America      3 days  
2218         NaN         NaN      Yes       Central US      7 days  
2219         NaN         NaN      Yes     North Africa      4 days  

[2220 rows x 29 columns]

In [24]:
# coverting the days_taken variable data type from string to integer
merged_df['days_taken'] = merged_df['days_taken'].dt.days

In [25]:
# calculating number of orders delivered on same day
merged_df[merged_df['days_taken']== 0 ].count().head(1)

Row ID    115
dtype: int64

In [26]:
# calculating number of orders delivered on 7th day
merged_df[merged_df['days_taken'] == 7].count().head(1)

Row ID    125
dtype: int64

In [27]:
# calculating number of orders delivered less than 7 days
merged_df[merged_df['days_taken'] <7].count().head(1)

Row ID    2095
dtype: int64

In [28]:
# calculating number of orders delivered on 15th day
merged_df[merged_df['days_taken'] ==15].count().head(1)

Row ID    0
dtype: int64

In [29]:
# calculating number of orders delivered less than 15 days
merged_df[merged_df['days_taken'] <15].count().head(1)

Row ID    2220
dtype: int64

In [30]:
# calculating number of orders delivered more than 30 days
merged_df[merged_df['days_taken'] >=30 ].count().head(1)

Row ID    0
dtype: int64

### QUESTION - 6 Run correlation check is there any relation between price and quantity

In [31]:
from scipy.stats import pearsonr

In [32]:
# inserting new variable unit price
merged_df['unit_price'] = merged_df[ 'Sales']/merged_df['Quantity']
merged_df

Row ID                  Order ID Order Date  Ship Date       Ship Mode  \
0      30191  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
1      30190  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
2      30187  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
3      30193  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
4      30189  IN-2012-PB19210127-41259 2012-12-16 2012-12-19     First Class   
...      ...                       ...        ...        ...             ...   
2215   35040  CA-2014-WB21850140-41654 2014-01-15 2014-01-21  Standard Class   
2216   35038  CA-2014-WB21850140-41654 2014-01-15 2014-01-21  Standard Class   
2217      25   US-2013-SJ2021582-41543 2013-09-26 2013-09-29     First Class   
2218   36974  CA-2012-ZC21910140-41271 2012-12-28 2013-01-04  Standard Class   
2219   41315    EG-2014-PM894038-41846 2014-07-26 2014-07-30  Standard Class   

       Customer ID     Customer Name    Segment  Postal Code           City  \
0      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
1      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
2      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
3      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
4      PB-19210127    Phillip Breyer  Corporate          NaN         Taipei   
...            ...               ...        ...          ...            ...   
2215  WB-218501406     William Brown   Consumer      10011.0  New York City   
2216  WB-218501406     William Brown   Consumer      10011.0  New York City   
2217    SJ-2021582      Sarah Jordon   Consumer          NaN     Hermosillo   
2218  ZC-219101402  Zuschuss Carroll   Consumer      60610.0        Chicago   
2219     PM-894038    Paul MacIntyre   Consumer          NaN     Alexandria   

      ... Discount    Profit Shipping Cost Order Priority Unnamed: 24  \
0     ...      0.0  720.3600       725.570       Critical         NaN   
1     ...      0.0  153.7500       627.270       Critical         NaN   
2     ...      0.0  311.8500       458.970       Critical         NaN   
3     ...      0.0  114.6000       203.920       Critical         NaN   
4     ...      0.0   68.2200        25.480       Critical         NaN   
...   ...      ...       ...           ...            ...         ...   
2215  ...      0.2   17.3376         2.670         Medium         NaN   
2216  ...      0.0    7.5992         1.690         Medium         NaN   
2217  ...      0.4  -75.5640         1.085         Medium         NaN   
2218  ...      0.6  -50.6688         5.290            Low         NaN   
2219  ...      0.0    9.0300         1.050         Medium         NaN   

     Unnamed: 25 Returned         Region_r  days_taken  unit_price  
0            NaN      Yes     Eastern Asia           3     857.580  
1            NaN      Yes     Eastern Asia           3     439.500  
2            NaN      Yes     Eastern Asia           3     452.010  
3            NaN      Yes     Eastern Asia           3     176.430  
4            NaN      Yes     Eastern Asia           3      49.440  
...          ...      ...              ...         ...         ...  
2215         NaN      Yes       Eastern US           6      16.512  
2216         NaN      Yes       Eastern US           6       4.130  
2217         NaN      Yes  Central America           3      20.124  
2218         NaN      Yes       Central US           7      12.992  
2219         NaN      Yes     North Africa           4      25.830  

[2220 rows x 30 columns]

In [33]:
# correlation between 
r = pearsonr(merged_df['unit_price'],merged_df['Quantity'])
r

PearsonRResult(statistic=-0.028330344451430955, pvalue=0.18208808203216745)

### A p-value greater than 0.05 suggests that the result is not statistically significant, meaning the observed correlation could be due to random chance.

### QUESTION- 7 Use the calculation from question - 5 and order priority to check is there any relationship between order priority and days_taken

In [34]:
from scipy.stats import pearsonr

In [35]:
# creating the function in order to decode the string values to integers
def recode(value):
    if value == 'Low':
        return 0
    elif value == 'Medium':
        return 1
    elif value == 'High':
        return 2
    else:
        return 3

In [36]:
# applying the decode to the variable
merged_df['order_priority_1'] = merged_df['Order Priority'].apply(recode)
merged_df.dtypes

Row ID                       int64
Order ID                    object
Order Date          datetime64[ns]
Ship Date           datetime64[ns]
Ship Mode                   object
Customer ID                 object
Customer Name               object
Segment                     object
Postal Code                float64
City                        object
State                       object
Country                     object
Region                      object
Market                      object
Product ID                  object
Category                    object
Sub-Category                object
Product Name                object
Sales                      float64
Quantity                     int64
Discount                   float64
Profit                     float64
Shipping Cost              float64
Order Priority              object
Unnamed: 24                float64
Unnamed: 25                float64
Returned                    object
Region_r                    object
days_taken          

In [37]:
# conducting the correlation
r = pearsonr(merged_df['order_priority_1'],merged_df['days_taken'])
r

PearsonRResult(statistic=-0.6132660296166021, pvalue=1.6871131006942964e-229)

#### as p-value is less than 0.05 therefore there is moderate negative correlation between the two variables, and this relationship is statistically significant 